src/ports/server/component.ts

```diff
@@ -315,6 +318,11 @@ export async function createServerComponent({
 
               const outcomeMessage: OutcomeResponseMessage = msg
               storedSocket.emit(MessageType.OUTCOME, outcomeMessage)
+
+              if (request.method === METHOD_DCL_PERSONAL_SIGN && request.sender) {
+                referral.updateReferral(request.sender)
+              }
```

Comment:
Although sender is sent to the server, we might need to verify that the sent signature belongs the sender, as it can be faked.